### PR TITLE
virttest.guest_agent: Fix timeout when executing powerdown test

### PR DESCRIPTION
**Pull request by xiagao**
Add check whether vm is alive when getting response from guest agent socket.
id: 1646695
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

File: virttest/guest_agent.py [hunk 1]
```diff
@@ -211,7 +211,8 @@ def _read_objects(self, timeout=READ_OBJECTS_TIMEOUT):
             return []
         s = b""
         end_time = time.time() + timeout
-        while self._data_available(end_time - time.time()):
+        while self._data_available(end_time - time.time()) and \
```
Comment (xutian) [inline virttest/guest_agent.py]: Please check verify Vm is alive before send guest-agent cmd

Comment (xutian) [reply virttest/guest_agent.py]: check verify VM is alive before send cmd in  function cmd()

**Comment by xiagao** [comment]
@luckyh  could you help check it?
Thanks..

**Comment by xiagao** [comment]
closed as there is another PR(https://github.com/avocado-framework/avocado-vt/pull/1857) can resolve this kind of problem.

**Pull request closed without merging**